### ISSUE TITLE
Sync more hash-url-params for iD

### ISSUE THREAD

**Issue by tordans** [openstreetmap/openstreetmap-website#3031]
This is the corresponding issue to https://github.com/openstreetmap/iD/issues/8114 in the iD Repo. This the solution will – as far as I understand – be part of this repos code, the topic should (also) be discussed here IMO.

I looked into this issue and will create a draft-PR after this Issue.

Short summary of the issue:

iD has 14 url params (+ the location/zoom params) that allow to configure the editor via URL.
Those params work fine to initialise iD on osm.org but get removed right after the editor loaded.
(This is https://github.com/openstreetmap/openstreetmap-website/blob/master/app/assets/javascripts/edit/id.js.erb#L23-L36)
Changes in the editor config are not represented in the hash URL on osm.org.
It is therefore not possible to share the configuration of an iD setup (apart from location+zoom) via URL.

Ideal solution:

The iD Dev-preview shows a different behaviour: Changes to the editor config are represented in the hash-URL.
Ideally, this behaviour is also possible for OSM.org (where we need to work around the iframe-setup).

**Comment by tomhughes** [comment]
If changes to the iD integration are needed then the iD authors normally provide them as part of the PR when they do a new release.

**Comment by tomhughes** [comment]
As a general point we don't expect people to deep link to iD on osm.org beyond simply copying the standard edit link - we don't really want to document or support secret URL parameters that people have to add manually because that's not a good user experience.

**Comment by tordans** [comment]
> …we don't expect people to deep link to iD on osm.org beyond simply copying the standard edit link - we don't really want to document or support secret URL parameters that people have to add manually because that's not a good user experience.

I agree and this is what this issue is about. Right now, the URL parameter are not documented and need do be added manually. They only work to initialise the editor but then disappear from the URL.

Once fixed, a user can use iD without knowing anything about the URLs since it updates by itself and represents a shareable configuration of what the user sees at this point (eg. which map layer is active, which mapillary filter is active and such).

**Comment by gravitystorm** [comment]
My concern with this is that "[cool URIs don't change](https://www.w3.org/Provider/Style/URI)". We've supported directly linking to the default editor for years, and providing a way to link to the editor at a certain position is something that is both useful, and will continue to make sense as we change editors (P1 -> P2 -> iD -> ???). 

But if we start exposing further options, do we need to try to support those links as the software changes? If someone starts embedding links in external websites to "edit here with this layer in the background and this menu open" then we risk having to support those links in perpetuity, or risk getting issues with "Our links to iD aren't loading the right background any more", etc - either because iD changed, or we changed editors, or whatever.

I do think there's a difference between what you can type into the URL bar (if you know the secret incantations) and what we show in the URL bar as a sort of promise that such a link will keep working. But perhaps I'm overthinking this one?
